tests/rptest/tests/datalake/datalake_usage_test.py

```diff
@@ -0,0 +1,190 @@
+                       timeout_sec=30,
+                       backoff_sec=2,
+                       err_msg=f"Timed out waiting for all usages to be empty")
+
+        def check_all_usages_atleast(
+                min_bytes: int,
+                prev_usages: Optional[dict[str, TopicUsage]] = None):
+            """
+            Check if all usages have kafka bytes processed greater than or equal to min_bytes.
+            """
+            def do_check():
+                usages = self.usage_by_topic()
+
+                usages_increased = True
+                if prev_usages is not None:
+                    usages_increased = all([
+                        (topic not in prev_usages.keys())
+                        or usage.kafka_bytes_processed
+                        > prev_usages[topic].kafka_bytes_processed
+                        for topic, usage in usages.items()
+                    ])
+
+                return usages_increased and all(
+                    usage.kafka_bytes_processed >= min_bytes
+                    for usage in usages.values())
+
+            wait_until(
+                do_check,
+                timeout_sec=30,
+                backoff_sec=2,
+                err_msg=
+                f"Timed out waiting for all usages to be at least {min_bytes} bytes"
+            )
+
+        def check_usage(usage: dict[str, TopicUsage]):
+            def do_check():
+                current_usages = self.usage_by_topic()
+                if len(current_usages) != len(usage):
+                    return False
+                for topic, expected_usage in usage.items():
+                    if topic not in current_usages:
+                        return False
+                    current_usage = current_usages[topic]
+                    if current_usage.revision != expected_usage.revision or current_usage.kafka_bytes_processed != expected_usage.kafka_bytes_processed:
+                        return False
+                return True
+
+            wait_until(do_check,
+                       timeout_sec=30,
+                       backoff_sec=2,
+                       err_msg=f"Timed out waiting for usage to match {usage}")
+
+        with DatalakeServices(self.test_ctx,
+                              redpanda=self.redpanda,
+                              include_query_engines=[query_engine],
+                              catalog_type=catalog_type) as dl:
+
+            # No usage before producing messages
+            check_no_usage()
+            for topic in ib_topics:
+                dl.create_iceberg_enabled_topic(topic,
+                                                partitions=3,
+                                                replicas=3,
+                                                target_lag_ms=10000)
+            # Produce some data and check usage.
+            produce_data(iteration=1)
+            check_all_usages_atleast(total_bytes)
+            current_usages = self.usage_by_topic()
```

Comment:
nit: maybe worth adding a sanity check somewhere that the number of topics in the report matches `len(ib_topics)`?